xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -764,28 +763,34 @@
       #set ($matchType = 'partial')
     #end
   #end
-  #set ($filterQuery = "#livetable_getFilterQuery(""${safe_tableAlias}.value"" $matchType)")
+  #set ($filterQuery = "#livetable_getFilterQuery(""${safe_tableAlias}.value"" $matchType false $multilistsize)")
   #set ($whereSql = "${whereSql} and $filterQuery.trim()")
-  #livetable_addFilterParam($filterValue $matchType $whereParams)
+  #livetable_addFilterParam($filterValue $matchType $whereParams $multilistsize)
 #end
 
 
-#macro (livetable_getFilterQuery $column $matchType $isList)
+#macro (livetable_getFilterQuery $column $matchType $isList $multilistsize)
   #if ($matchType == 'partial' || $matchType == 'prefix')
     upper($column) like upper(?)
   #elseif ($isList)
     ? in elements($column)
+  #elseif (($multilistsize &gt; 1) &amp;&amp; ($matchType == 'exact'))
+    $column in ( $stringtool.repeat('?', ', ', $multilistsize) )
   #else
     $column = ?
   #end
 #end
 
 
-#macro (livetable_addFilterParam $filterValue $matchType $params)
+#macro (livetable_addFilterParam $filterValue $matchType $params $multilistsize)
```

Comment:
From what I can see ``livetable_addFilterParam`` only adds parameters to the ``$params`` so no AND/OR is needed. Calling ``livetable_addFilterParam`` in a loop should add multiple parameters to ``$params``. When I said

> do the ``#foreach`` outside

I meant right when ``livetable_addFilterParam`` is called. Now, in order to not do the ``#foreach`` loop every time ``livetable_addFilterParam`` is called, we could define ``livetable_addFilterParams`` (plural).